src/spac/spatial_analysis.py

```diff
@@ -192,42 +264,49 @@ def Cluster_Interaction_Matrix_Analysis(
                 ] == subset_key
             ].copy()
 
-            sq.gr.spatial_neighbors(subset_adata)
-
             buffer.seek(0)
 
             ax_copy = pickle.load(buffer)
 
-            if analysis_method == "Neighborhood Enrichment":
-                ax_copy = Neighborhood_Enrichment_Analysis(
-                        subset_adata,
-                        new_annotation_name,
-                        ax_copy)
-
-            elif analysis_method == "Cluster Interaction Matrix":
-                ax_copy = Cluster_Interaction_Matrix_Analysis(
-                        subset_adata,
-                        new_annotation_name,
-                        ax_copy)
-
-            ax_dictionary[subset_key] = ax_copy
+            ax_copy = perform_analysis(
+                            subset_adata,
+                            analysis_method,
+                            new_annotation_name,
+                            ax_copy,
+                            return_matrix
+                        )
+
+            if return_matrix:
+                ax_dictionary[subset_key] = ax_copy[0]
+                matrix_dictionary[subset_key] = ax_copy[1]
+            else:
+                ax_dictionary[subset_key] = ax_copy
 
             del subset_adata
 
-    else:
-        sq.gr.spatial_neighbors(adata)
+        if return_matrix:
+            results = [
+                {"Ax": ax_dictionary},
+                {"Matrix": matrix_dictionary}
+            ]
+        else:

```

Comment:
@ruiheesi, what about we make the behavior that we always return a dictionary. The dictionary can have the "Ax' key always, and optionally it would have the "Matrix" key.

Reply:
Hi George, we are already doing it. Thanks.